src/rules/color-named/index.js

```diff
@@ -55,6 +56,8 @@ export default function (expectation, options) {
       valueParser(decl.value).walk(node => {
         const { value, type, sourceIndex } = node
 
+        if (optionsMatches(options, "ignore", "inside-function") && type === "function") { return false }
```

Comment:
I think this needs to only return false when the type is a function *and* the function *is not* a colour representation like `rgb` or `gray` etc.